### ISSUE TITLE
Incorrect Panic Warning Location

### ISSUE THREAD

**Issue by stoneman258** [lizhuohua/rust-mir-checker#17]
## Description
When using the following case, the `mirchecker` reports an incorrect panic warning location. The correct location for the panic warning should be within the `if maybe && capacity != 0 block`, specifically at the `panic!("Unable to allocate requested capacity")` line. However, `mirchecker` reports the warning at the `maybe` line after the `if block`.

### Case
```rust
pub fn assert_val(maybe_wrong: std::sync::atomic::AtomicBool) {
    while {
        let maybe_wrong = maybe_wrong.load(std::sync::atomic::Ordering::SeqCst);
        if maybe_wrong {
            panic!()//should report warning
        }
        maybe_wrong//report warning
    } {}
}
```

## Expected Behavior
The panic warning should only be reported at line 5.

## Actual Behavior
`mirchecker` reports a panic warning at line 7.
